src/pudl/output/ferc1.py

```diff
@@ -1752,36 +1753,57 @@ def set_forest_attributes(
             data={"tags": list(tags_dict.values())},
         ).reset_index()
 
-        tags_dict = tags.set_index(["table_name", "xbrl_factoid"]).to_dict(
-            orient="index"
+        # There are a few rare instances where a particular node is specified with more
+        # than one weight (-1 vs. 1) and in those cases, we always want to keep the
+        # weight of -1, since it affects the overall root->leaf calculation outcome.
+        multi_valued_weights = (
+            self.exploded_calcs.groupby(self.calc_cols, dropna=False)["weight"]
+            .transform("nunique")
+            .gt(1)
+        )
+        calcs_to_drop = multi_valued_weights & (self.exploded_calcs.weight == 1)
+        # Check that there are only a *few* multi-valued weights.
+        mv_wt_frac = sum(multi_valued_weights) / len(self.exploded_calcs)
+        for table in self.table_names:
+            if table in MAX_MULTIVALUE_WEIGHT_FRAC:
+                max_mv_wt_frac = MAX_MULTIVALUE_WEIGHT_FRAC[table]
+
+        logger.info(
+            f"Found {sum(multi_valued_weights)}/{len(self.exploded_calcs)} "
+            f"({mv_wt_frac:.2%}; max allowed: {max_mv_wt_frac:.2%}) "
+            "nodes having conflicting weights. Dropping "
+            f"{sum(calcs_to_drop)/len(self.exploded_calcs):.2%} where weight == 1"
         )
```

Comment:
I think it needs to be down here, and may or may not be useful earlier in the process, because in the end what we care about is whether there are *conflicting* weights, and only in the nodes that remain in the tree after we've done all the pruning, which is just a subset of the records in the original `exploded_calcs`.  I left a quieter, less extensive logging message up in `single_valued_weights()` just for informational purposes.

My intuition is this shouldn't be validated using a pydantic validator because it's quite a complicated derived value, rather than one of the inputs, which we need to check for  validity.